gap/SnAnBB.gi

```diff
@@ -18,100 +18,906 @@
 ##  The code is based upon the algorithm presented in the paper [BLGN+03].
 ##
 ##
-# DeclareInfoClass( "InfoRecSnAn" );
-# SetInfoLevel( InfoRecSnAn, 0 );
-#
-# # The following are used for comparisons:
```

Comment:
In this file, did you only uncomment lines, or did you also perform changes afterwards?

Related to this: If you think you are basically done with this PR, perhaps you could now clean up its history.  IMHO it is not very useful right now (actually, to me, it seems completely useless, but you might disagree; I am happy to discuss this :-).

What I'd do is probably to is this:
1. make sure latest `master` and latest version of this branch are present
2. create a temporary branch which just undoes the commenting out of this file:
    ```
    git checkout -b HACK master 
    perl -pi -e 's/^# //' gap/SnAnBB.gi # undo the commenting out
    git commit -m "Undo commenting out of unused code in gap/SnAnBB.gi" gap/SnAnBB.gi
    ```
    Actually, having done this, you probably want to do a variant of this, were you leave `RecogniseSnAn` and `NiceGeneratorsSnAn` commented out; and then in another commit (before or after this one), move `NiceGeneratorsSnAn` to the end of the file, to match what you did in this PR.
3. switch to `frss/AnSnRecogUnknownDegree`, make sure it is a superset of master:
    ```
    git checkout frss/AnSnRecogUnknownDegree
    git merge master
    ```
4. now discard the useless history and start over, based on the commit we just made in the HACK branch:
    ```
    git reset HACK
    git add doc gap tst read.g
    git commit
    ```

To make sure you both get credit, add `Co-authored-by: NAME <EMAIL>` to the end of the commit message.

With that, it becomes easy to see what you changed in `gap/SnAnBB.gi`. I just followed the steps above and it was quite helpful, and would also help for reviewing this PR.

Now, of course the above can be further refined; e.g. you might want to split the commits further. But this is up to you.